acbuild/annotation.go

```diff
@@ -83,16 +96,29 @@ func runRmAnno(cmd *cobra.Command, args []string) (exit int) {
 		return 1
 	}
 
+	err := getLock()
+	if err != nil {
+		stderr("annotation remove: %v", err)
+		return 1
+	}
+
 	if debug {
 		stderr("Removing annotation %q", args[0])
 	}
 
-	err := lib.RemoveAnnotation(tmpacipath(), args[0])
+	err = lib.RemoveAnnotation(tmpacipath(), args[0])
 
+	var retcode int
 	if err != nil {
 		stderr("annotation remove: %v", err)
-		return 1
+		retcode = 1
```

Comment:
> Not removing the lockfile is pretty irrelevant to acbuild, but from a UX standpoint I suppose the user would expect the file to not exist when acbuild doesn't have a lock on it. 

Right I think it would be cleaner to remove the file in the usual case

> I'd also prefer to not defer the release, since if the lock can't be released for whatever reason I'd really like to emit an error.

I meant deferring a closure/inline func, not just the releaseLock() call. You can still emit and propagate errors.
 
> I suppose there's a chance someone (possibly myself when writing tests) would directly call the cmd* functions, and thus couldn't rely on the lock being released when the process exits.

I think it's perfectly reasonable to not support running the cmd functions directly from code, they should be the CLI glue and may do exits/direct stdout/stderr IO/blabla